app/models/uploaded_file.rb

```diff
@@ -4,15 +4,21 @@ class UploadedFile < ActiveRecord::Base
 
   def self.create_and_resize(attrs)
     tempfile_path = attrs[:file].tempfile.path
-    unless attrs[:content_type] == 'application/pdf'
-      if attrs[:content_type] == 'image/png'
-        # convert png -> jpeg
-        image = MiniMagick::Image.new(tempfile_path)
-        image.format 'jpg'
-        attrs[:content_type] = 'image/jpeg'
+    begin
+      unless attrs[:content_type] == 'application/pdf'
+        if attrs[:content_type] == 'image/png'
+          # convert png -> jpeg
+          image = MiniMagick::Image.new(tempfile_path)
+          if image.valid?
```

Comment:
@kaptron did you implement the conversion/optimization feature? I actually have a Q about this, how come we don't use the image variable again past conversion and use tempfile_path?

Reply:
@chrisdolendo ah -- `image` is just a MiniMagick object that represents the converted image that lives at `tempfile_path` (and has access to MiniMagick methods like `.valid?` and `.format(xxx)`. So once you do the conversion, the image at `tempfile_path` has been modified, and so when you want to refer to the actual file in the filesystem, you refer to that.

Reply:
makes sense thanks @kaptron!